Server crashes when polishing on the wrong grindstone

Using the grindstone to polish items causes a crash, probably an incompability

Fabric 
Minecraft version: 1.20.1
Spelunkery version: 0.3.6
Moonlight Lib version: 2.12.11
Fabric API / QSL / Forge version: 0.16

**Other mods:**
Fabric Mods: 
		amendments: Amendments 1.20-1.2.8
		another_furniture: Another Furniture 1.20.1-3.0.1
		appleskin: AppleSkin 2.5.1+mc1.20
		architectury: Architectury 9.2.14
		athena: Athena 3.1.2
		balm-fabric: Balm 7.3.7
			kuma_api: KumaAPI 20.1.8
		bclib: BCLib 3.0.14
			wunderlib: WunderLib 1.1.5
		beautify: Beautify 1.2.0+1.20.1
		betterarcheology: Better Archeology 1.1.9-1.20.1
		betterdeserttemples: YUNG's Better Desert Temples 1.20-Fabric-3.0.3
			org_reflections_reflections: reflections 0.10.2
		betterdungeons: YUNG's Better Dungeons 1.20-Fabric-4.0.4
		betterend: Better End 4.0.11
		betterfortresses: YUNG's Better Nether Fortresses 1.20-Fabric-2.0.6
		betterjungletemples: YUNG's Better Jungle Temples 1.20-Fabric-2.0.5
		bettermineshafts: YUNG's Better Mineshafts 1.20-Fabric-4.0.4
		betternether: Better Nether 9.0.10
		betteroceanmonuments: YUNG's Better Ocean Monuments 1.20-Fabric-3.0.4
		betterstrongholds: YUNG's Better Strongholds 1.20-Fabric-4.0.3
		betterwitchhuts: YUNG's Better Witch Huts 1.20-Fabric-3.0.3
		biolith: Biolith 1.0.0-beta.2
			terraform-surfaces-api-v1: Terraform Surfaces API (v1) 7.0.2
		biomeswevegone: Oh The Biomes We've Gone 1.0.6-Beta
		boids: Boids 1.2.2
		bookshelf: Bookshelf 20.2.13
		bosses_of_mass_destruction: Bosses of Mass Destruction (Beta) 1.7.5-1.20.1
			maelstrom_library: Maelstrom Library 1.6.1-1.20
			multipart_entities: MultipartEntities 1.5-1.20
		botarium: Botarium 2.3.4
			team_reborn_energy: Energy 3.0.0
		bountiful: Bountiful 6.0.3+1.20.1
		c2me: Concurrent Chunk Management Engine 0.2.0+alpha.11.13
			c2me-base: Concurrent Chunk Management Engine (Base) 0.2.0+alpha.11.13
			c2me-fixes-chunkio-threading-issues: Concurrent Chunk Management Engine (Fixes/Chunk IO/Threading Issues) 0.2.0+alpha.11.13
			c2me-fixes-general-threading-issues: Concurrent Chunk Management Engine (Fixes/General/Threading Issues) 0.2.0+alpha.11.13
			c2me-fixes-worldgen-threading-issues: Concurrent Chunk Management Engine (Fixes/WorldGen/Threading Issues) 0.2.0+alpha.11.13
			c2me-fixes-worldgen-vanilla-bugs: Concurrent Chunk Management Engine (Fixes/WorldGen/Vanilla Bugs) 0.2.0+alpha.11.13
			c2me-notickvd: Concurrent Chunk Management Engine (No Tick View Distance) 0.2.0+alpha.11.13
			c2me-opts-allocs: Concurrent Chunk Management Engine (Optimizations/Memory Allocations) 0.2.0+alpha.11.13
			c2me-opts-chunk-access: Concurrent Chunk Management Engine (Optimizations/Chunk Access) 0.2.0+alpha.11.13
			c2me-opts-chunkio: Concurrent Chunk Management Engine (Optimizations/Chunk IO) 0.2.0+alpha.11.13
			c2me-opts-math: Concurrent Chunk Management Engine (Optimizations/Math) 0.2.0+alpha.11.13
			c2me-opts-scheduling: Concurrent Chunk Management Engine (Optimizations/Scheduling) 0.2.0+alpha.11.13
			c2me-opts-worldgen-general: Concurrent Chunk Management Engine (Optimizations/General WorldGen) 0.2.0+alpha.11.13
			c2me-opts-worldgen-vanilla: Concurrent Chunk Management Engine (Optimizations/Vanilla WorldGen) 0.2.0+alpha.11.13
			c2me-rewrites-chunk-serializer: Concurrent Chunk Management Engine (Rewrites/Chunk Serializer) 0.2.0+alpha.11.13
			c2me-rewrites-chunkio: Concurrent Chunk Management Engine (Rewrites/Chunk IO) 0.2.0+alpha.11.13
			c2me-server-utils: Concurrent Chunk Management Engine (Server Utils) 0.2.0+alpha.11.13
			c2me-threading-chunkio: Concurrent Chunk Management Engine (Threading/WorldGen) 0.2.0+alpha.11.13
			c2me-threading-lighting: Concurrent Chunk Management Engine (Threading/Lighting) 0.2.0+alpha.11.13
			c2me-threading-worldgen: Concurrent Chunk Management Engine (Threading/WorldGen) 0.2.0+alpha.11.13
			com_ibm_async_asyncutil: asyncutil 0.1.0
			net_objecthunter_exp4j: exp4j 0.4.8
			org_threadly_threadly: threadly 7.0
		cardinal-components: Cardinal Components API 5.2.2
			cardinal-components-base: Cardinal Components API (base) 5.2.2
			cardinal-components-block: Cardinal Components API (blocks) 5.2.2
			cardinal-components-chunk: Cardinal Components API (chunks) 5.2.2
			cardinal-components-entity: Cardinal Components API (entities) 5.2.2
			cardinal-components-item: Cardinal Components API (items) 5.2.2
			cardinal-components-level: Cardinal Components API (world saves) 5.2.2
			cardinal-components-scoreboard: Cardinal Components API (scoreboard) 5.2.2
			cardinal-components-world: Cardinal Components API (worlds) 5.2.2
		charmofundying: Charm of Undying 6.5.0+1.20.1
		chefsdelight: Chefs Delight 1.0.3-fabric-1.20.1
		chipped: Chipped 3.0.6
		chunky: Chunky 1.3.146
		cicada: CICADA 0.8.3+1.20.1
		cloth-config: Cloth Config v11 11.1.118
			cloth-basic-math: cloth-basic-math 0.6.1
		clumps: Clumps 12.0.0.4
		collective: Collective 7.81
		comforts: Comforts 6.4.0+1.20.1
			spectrelib: SpectreLib 0.13.15+1.20.1
		connectiblechains: Connectible Chains 2.2.1+1.20.1
		corgilib: CorgiLib 4.0.2.0
			com_electronwill_night-config_core: core 3.8.0
			com_electronwill_night-config_toml: toml 3.8.0
		crafttweaker: CraftTweaker 14.0.40
		create: Create 0.5.1-f-build.1417+mc1.20.1
			com_google_code_findbugs_jsr305: jsr305 3.0.2
			milk: Milk Lib 1.2.60
				dripstone_fluid_lib: Dripstone Fluid Lib 3.0.2
			porting_lib_brewing: Porting Lib Brewing 2.3.2+1.20.1
			registrate-fabric: Registrate for Fabric 1.3.62-MC1.20.1
				porting_lib_tags: Porting Lib Tags 3.0
					porting_lib_gametest: Porting Lib GameTest 2.1.1090+1.20
		create-universal-ores: Crate Universal Ores 1.0.0-fabric-1.20.1
		create_dd: Create: Dreams & Desires 0.1d
			dsbg: Detected setBlock Be Gone 1.0
		create_structures: Create: Structures 1.1.0
		createaddition: Create Crafts & Additions 1.2.4
		createdeco: Create Deco 2.0.2-1.20.1-fabric
		creeperoverhaul: Creeper Overhaul 3.0.2
		cristellib: Cristel Lib 1.1.5
		ctov: ChoiceTheorem's Overhauled Village 3.4.6
		cutthrough: Cut Through 8.0.2
		dawn: Dawn API 5.0.0
			terraform-shapes-api-v1: Terraform Shapes API (v1) 7.0.1
			terraform-wood-api-v1: Terraform Wood API (v1) 7.0.1
		deuf_refabricated: DEUF Refabricated 1.1.0
		diagonalfences: Diagonal Fences 8.1.4
			diagonalblocks: Diagonal Blocks 8.0.5
		do_a_barrel_roll: Do a Barrel Roll 3.6.1+1.20.1
			fabric-permissions-api-v0: fabric-permissions-api 0.2-SNAPSHOT
		doapi: [Let's Do] API 1.2.15
		elytraslot: Elytra Slot 6.4.2+1.20.1
			mixinsquared: MixinSquared 0.1.2-beta.6
		emi: EMI 1.1.12+1.20.1+fabric
		endermanoverhaul: Enderman Overhaul 1.0.4
		endrem: End Remastered 5.2.4
		ends_delight: End's Delight 2.1
			porting_lib_config: Porting Lib Config 2.3.4+1.20.1
		entityculling: EntityCulling 1.6.6
		everycomp: Every Compat 1.20-2.6.71
			porting_lib_accessors: Porting Lib Accessors 2.3.4+1.20.1
			porting_lib_attributes: Porting Lib Attributes 2.3.4+1.20.1
				porting_lib_core: Porting Lib Core 2.3.4+1.20.1
			porting_lib_base: Porting Lib Base 2.3.4+1.20.1
				porting_lib_utility: Porting Lib Utility 2.3.4+1.20.1
			porting_lib_common: Porting Lib Common 2.3.4+1.20.1
			porting_lib_entity: Porting Lib Entity 2.3.4+1.20.1
				porting_lib_mixin_extensions: Porting Lib Mixin Extensions 2.3.4+1.20.1
			porting_lib_extensions: Porting Lib Extensions 2.3.4+1.20.1
			porting_lib_lazy_registration: Porting Lib Lazy Register 2.3.4+1.20.1
			porting_lib_model_generators: Porting Lib Model Generators 2.3.4+1.20.1
				porting_lib_data: Porting Lib Data 2.3.4+1.20.1
				porting_lib_model_materials: Porting Lib Model Materials 2.3.4+1.20.1
			porting_lib_model_loader: Porting Lib Model Loader 2.3.4+1.20.1
			porting_lib_models: Porting Lib Models 2.3.4+1.20.1
				porting_lib_fluids: Porting Lib Fluids 2.3.4+1.20.1
			porting_lib_networking: Porting Lib Networking 2.3.4+1.20.1
			porting_lib_obj_loader: Porting Lib Obj Loader 2.3.4+1.20.1
			porting_lib_transfer: Porting Lib Transfer 2.3.4+1.20.1
		expandeddelight: Expanded Delight 0.3.1
			omega-config: OmegaConfig 1.4.0+1.20.1
		fabric-api: Fabric API 0.92.2+1.20.1
			fabric-api-base: Fabric API Base 0.4.31+1802ada577
			fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.6.36+1802ada577
			fabric-biome-api-v1: Fabric Biome API (v1) 13.0.13+1802ada577
			fabric-block-api-v1: Fabric Block API (v1) 1.0.11+1802ada577
			fabric-block-view-api-v2: Fabric BlockView API (v2) 1.0.1+1802ada577
			fabric-command-api-v1: Fabric Command API (v1) 1.2.34+f71b366f77
			fabric-command-api-v2: Fabric Command API (v2) 2.2.13+1802ada577
			fabric-commands-v0: Fabric Commands (v0) 0.2.51+df3654b377
			fabric-containers-v0: Fabric Containers (v0) 0.1.64+df3654b377
			fabric-content-registries-v0: Fabric Content Registries (v0) 4.0.11+1802ada577
			fabric-convention-tags-v1: Fabric Convention Tags 1.5.5+1802ada577
			fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.2.19+1802ada577
			fabric-data-attachment-api-v1: Fabric Data Attachment API (v1) 1.0.0+de0fd6d177
			fabric-data-generation-api-v1: Fabric Data Generation API (v1) 12.3.4+1802ada577
			fabric-dimensions-v1: Fabric Dimensions API (v1) 2.1.54+1802ada577
			fabric-entity-events-v1: Fabric Entity Events (v1) 1.6.0+1c78457f77
			fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.6.2+1802ada577
			fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.63+df3654b377
			fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.40+1802ada577
			fabric-item-api-v1: Fabric Item API (v1) 2.1.28+1802ada577
			fabric-item-group-api-v1: Fabric Item Group API (v1) 4.0.12+1802ada577
			fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 2.2.22+1802ada577
			fabric-loot-api-v2: Fabric Loot API (v2) 1.2.1+1802ada577
			fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.1.45+9e7660c677
			fabric-message-api-v1: Fabric Message API (v1) 5.1.9+1802ada577
			fabric-mining-level-api-v1: Fabric Mining Level API (v1) 2.1.50+1802ada577
			fabric-networking-api-v1: Fabric Networking API (v1) 1.3.11+1802ada577
			fabric-networking-v0: Fabric Networking (v0) 0.3.51+df3654b377
			fabric-object-builder-api-v1: Fabric Object Builder API (v1) 11.1.3+1802ada577
			fabric-particles-v1: Fabric Particles (v1) 1.1.2+1802ada577
			fabric-recipe-api-v1: Fabric Recipe API (v1) 1.0.21+1802ada577
			fabric-registry-sync-v0: Fabric Registry Sync (v0) 2.3.3+1802ada577
			fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.3.37+92a0d36777
			fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 3.0.28+1802ada577
			fabric-resource-conditions-api-v1: Fabric Resource Conditions API (v1) 2.3.8+1802ada577
			fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.11.10+1802ada577
			fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.3.30+1802ada577
			fabric-transfer-api-v1: Fabric Transfer API (v1) 3.3.5+8dd72ea377
			fabric-transitive-access-wideners-v1: Fabric Transitive Access Wideners (v1) 4.3.1+1802ada577
		fabric-language-kotlin: Fabric Language Kotlin 1.12.0+kotlin.2.0.10
			org_jetbrains_kotlin_kotlin-reflect: kotlin-reflect 2.0.10
			org_jetbrains_kotlin_kotlin-stdlib: kotlin-stdlib 2.0.10
			org_jetbrains_kotlin_kotlin-stdlib-jdk7: kotlin-stdlib-jdk7 2.0.10
			org_jetbrains_kotlin_kotlin-stdlib-jdk8: kotlin-stdlib-jdk8 2.0.10
			org_jetbrains_kotlinx_atomicfu-jvm: atomicfu-jvm 0.25.0
			org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm: kotlinx-coroutines-core-jvm 1.8.1
			org_jetbrains_kotlinx_kotlinx-coroutines-jdk8: kotlinx-coroutines-jdk8 1.8.1
			org_jetbrains_kotlinx_kotlinx-datetime-jvm: kotlinx-datetime-jvm 0.6.0
			org_jetbrains_kotlinx_kotlinx-io-bytestring-jvm: kotlinx-io-bytestring-jvm 0.5.1
			org_jetbrains_kotlinx_kotlinx-io-core-jvm: kotlinx-io-core-jvm 0.5.1
			org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm: kotlinx-serialization-cbor-jvm 1.7.1
			org_jetbrains_kotlinx_kotlinx-serialization-core-jvm: kotlinx-serialization-core-jvm 1.7.1
			org_jetbrains_kotlinx_kotlinx-serialization-json-jvm: kotlinx-serialization-json-jvm 1.7.1
		fabricloader: Fabric Loader 0.16.0
		farm_and_charm: [Let's Do] Farm & Charm 1.0.1
			blue_endless_jankson: jankson 1.2.3
		farmersdelight: Farmer's Delight 1.20.1-2.1.2+refabricated
			porting_lib_client_events: Porting Lib Client Events 2.3.4+1.20.1
			porting_lib_loot: Porting Lib Loot 2.3.4+1.20.1
			porting_lib_recipe_book_categories: Porting Lib Recipe Book Categories 2.3.4+1.20.1
			porting_lib_registries: Porting Lib Registries 2.3.4+1.20.1
			porting_lib_tool_actions: Porting Lib Tool Actions 2.3.4+1.20.1
		faux-custom-entity-data: Faux-Custom-Entity-Data 6.0.1
		featurerecycler: Feature Recycler 1.0.0
		ferritecore: FerriteCore 6.0.1
		forgeconfigapiport: Forge Config API Port 8.0.0
		framework: Framework 0.6.16
			org_javassist_javassist: javassist 3.29.2-GA
		friendsandfoes: Friends&Foes 2.0.16
		frightsdelight: Frights Delight fabric-1.20.1-1.0.4
		full_slabs: FullSlabs 3.4.0
			statement: Statement 4.2.8+1.14.4-1.20.1
				kanos_config: Kanos Config 0.4.1+1.14.4-1.19.4
				statement_vanilla_compatibility: Statement Vanilla Compatibility Module 1.0.1+1.16.5-1.17
		fwaystones: Fabric Waystones 3.3.2+mc1.20.1
		galosphere: Galosphere 1.20.1-1.4.1
		geckolib: GeckoLib 4 4.4.7
			com_eliotlash_mclib_mclib: mclib 20
		geophilic: Geophilic 3.1.1
		gipples_galore: Gipples Galore 1.0.0
		goblintraders: Goblin Traders 1.9.3
		goodending: Good Ending 1.20.1-1.0.1-fabric
		handcrafted: Handcrafted 3.0.6
		hearth_and_home: Hearth & Home 1.20.1-2.0.3
		icterine: Icterine 1.3.0
		illagerinvasion: Illager Invasion 8.0.5
			extensibleenums: Extensible Enums 7.0.1
		jade: Jade 11.10.0+fabric
		jadeaddons: Jade Addons 5.3.1+fabric
		java: OpenJDK 64-Bit Server VM 17
		kambrik: Kambrik 6.1.1+1.20.1
		krypton: Krypton 0.2.3
			com_velocitypowered_velocity-native: velocity-native 3.2.0-SNAPSHOT
		labels: labels 1.20-1.20.2
		lavender: Lavender 0.1.9+1.20
			lavender-md: lavender-md 0.1.1+1.20
			lavender-md-owo-ui: lavender-md-owo-ui 0.1.1+1.20
		lazydfu: LazyDFU 0.1.3
		leavesbegone: Leaves Be Gone 8.0.0
		lithium: Lithium 0.11.2
		lithostitched: Lithostitched 1.1.8
		mavapi: More Axolotl Variants API 1.1.4
		mavm: More Axolotl Variants Mod 1.2.6
		maxhealthfix: MaxHealthFix 12.0.3
		mc249136: MC-249136 (Fabric) 1.0.0
		memoryleakfix: Memory Leak Fix 1.1.5
		mes: Moog's End Structures 1.3.1-1.20-fabric
		midnightlib: MidnightLib 1.4.1
		minecraft: Minecraft 1.20.1
		mixintrace: MixinTrace 1.1.1+1.17
		modernfix: ModernFix 5.19.1+mc1.20.1
		moonlight: Moonlight 1.20-2.12.11
		morediscs: More Music Discs 1.20.1-33.1-fabric
		moremobvariants: More Mob Variants 1.3.1
		mr_ctov_beautifycompat: CTOV - Beautify Compat 2.0
		mr_ctov_chefsdelightcompat: CTOV - Chef's delight Compat 2.0
		mr_ctov_farmersdelightcompat: CTOV - Farmer Delight Compat 2.1
		mr_ctov_friendsandfoescompat: CTOV - Friends and Foes Compat 2.0
		mr_dungeons_andtaverns: Dungeons and Taverns 3.0.3.f
		mru: Mineblock's Repeated Utilities 0.4.5+1.20.1
		mythicupgrades: Mythic Upgrades 4.1.4+mc1.20.1
			mixinextras: MixinExtras 0.4.1
		naturescompass: Nature's Compass 1.20.1-2.2.3-fabric
		nerb: Not Enough Recipe Book 0.3
		netherdepthsupgrade: Nether Depths Upgrade fabric-3.1.6-1.20
		noisium: Noisium 2.2.2+mc1.20-1.20.1
		oceansdelight: Ocean's Delight fdrf-fabric-1.0.2-1.20
		ohthetreesyoullgrow: Oh The Trees You'll Grow 1.2.0
		owo: oωo 0.11.2+1.20
		packetfixer: Packet Fixer 1.4.2
		patchouli: Patchouli 1.20.1-84-FABRIC
			fiber: fiber 0.23.0-2
		phantasm: End's Phantasm 0.4
		polymorph: Polymorph 0.49.5+1.20.1
		promenade: Promenade 4.1.1
		puzzleslib: Puzzles Lib 8.1.21
			puzzlesaccessapi: Puzzles Access Api 8.0.7
		railways: Create: Steam 'n' Rails 1.6.4+fabric-mc1.20.1
			mm: Manningham Mills 2.3
		redirector: Redirector 1.20.1-4.4.0
		regions_unexplored: Regions Unexplored 0.5.6+1.20.1
		replantingcrops: Replanting Crops 5.4
		repurposed_structures: Repurposed Structures 7.1.15+1.20.1-fabric
		resourcefulconfig: Resourcefulconfig 2.1.2
		resourcefullib: Resourceful Lib 2.1.28
			com_teamresourceful_bytecodecs: bytecodecs 1.0.2
			com_teamresourceful_yabn: yabn 1.0.3
		reterraforged: ReTerraForged 0.0.6
		ribbits: Ribbits 1.20.1-Fabric-3.0.0
		sawmill: Universal Sawmill 1.20-1.4.3
		seasons: Fabric Seasons 2.3+1.20
		seasonsdelightcompat: Fabric Seasons: Delight Compat 1.0-1.3.9
		seasonsextras: Fabric Seasons: Extras 1.2-BETA+1.20
		seasonsterralithcompat: Fabric Seasons: Terralith Compat 1.0-2.3.7
		shulkerboxslot: Shulker Box Slot 5.1.1+1.20.1
		shulkerboxtooltip: Shulker Box Tooltip 4.0.4+1.20.1
		smallships: Small Ships 2.0.0-b1.4
		smarterfarmers: Smarter Farmers 1.20-1.8.2
		snifferplus: Hellion's Sniffer+ 0.3.0
		snuffles: Snuffles 1.0.0+1.20
		sophisticatedbackpacks: Sophisticated Backpacks 3.19.5+mc1.20.1-SNAPSHOT-build.104
		sophisticatedcore: Sophisticated Core 0.5.109+mc1.20.1-SNAPSHOT-build.104
		sophisticatedstorage: Sophisticated Storage 0.8.60+mc1.20.1-SNAPSHOT-build.104
		sound_physics_remastered: Sound Physics Remastered 1.20.1-1.4.5
		spawn: Spawn 1.0.3-fabric
		species: Species 1.3-fabric
		spelunkery: Spelunkery 1.20.1-0.3.6
		spiceoffabric: Spice of Fabric 1.6.2+mc1.20.1
			capsaicin: Capsaicin 1.3.3+mc1.20.1
			tweed4_annotated: tweed4_annotated 1.3.1+mc1.20-pre1
			tweed4_base: tweed4_base 1.7.1+mc1.20-pre1
			tweed4_data: tweed4_data 1.2.1+mc1.20-pre1
			tweed4_data_hjson: tweed4_data_hjson 1.1.1+mc1.20-pre1
			tweed4_tailor_lang_json_descriptions: tweed4_tailor_lang_json_descriptions 1.1.0+mc1.20-pre1
		squashed: Squashed 1.0.4+1.20.1
		starlight: Starlight 1.1.2+fabric.dbc156f
		supplementaries: Supplementaries 1.20-2.8.17
		suppsquared: Supplementaries Squared 1.20-1.1.14
		t_and_t: Towns and Towers 1.12
		tectonic: Tectonic 2.3.5b
		terrablender: TerraBlender 3.0.1.7
		terrablenderfix: TerraBlenderFix 0.0.1
		terralith: Terralith 2.5.4
		things: Things 0.3.3+1.20
		trinkets: Trinkets 3.7.2
		twigs: Twigs 3.1.0
		universal-graves: Universal Graves 3.0.2+1.20.1
			common-protection-api: Common Protection API 1.0.0
			placeholder-api: Placeholder API 2.1.3+1.20.1
			polymer-core: Polymer (Core) 0.5.19+1.20.1
				polymer-networking: Polymer (Networking) 0.5.19+1.20.1
				polymer-registry-sync-manipulator: Polymer (Registry Sync Manipulator) 0.5.19+1.20.1
			polymer-resource-pack: Polymer (Resource Pack) 0.5.19+1.20.1
				polymer-common: Polymer (Common) 0.5.19+1.20.1
			polymer-virtual-entity: Polymer (Virtual Entity) 0.5.19+1.20.1
			predicate-api: Predicate API 0.2.1+1.20.1
			server_translations_api: Server Translations API 2.0.0+1.20
				packet_tweaker: Packet Tweaker 0.4.0+1.19.4
			sgui: sgui 1.2.2+1.20
		universal_ores: Universal Ores 1.5.2
		variantsandventures: Variants & Ventures 1.0.5
		vcinteraction: Voice Chat Interaction 1.20.1-1.0.6
		veinmining: Vein Mining 1.5.0+1.20.1
		voicechat: Simple Voice Chat 1.20.1-2.5.20
		waterdripsound: Drip Sounds 1.19-0.3.2
		wilderwild: Wilder Wild 2.4.6-mc1.20.1
			frozenlib: FrozenLib 1.8-mc1.20.1
			reach-entity-attributes: Reach Entity Attributes 2.4.0
		wildlife: Wildlife 1.19.2-2.2
		xaerominimap: Xaero's Minimap 24.2.0
		xaeroworldmap: Xaero's World Map 1.38.8
		xxlpackets: XXLPackets 1.0.4-1.20.x-1.20.1
		yet_another_config_lib_v3: YetAnotherConfigLib 3.5.0+1.20.1-fabric
			com_twelvemonkeys_common_common-image: common-image 3.10.0
			com_twelvemonkeys_common_common-io: common-io 3.10.0
			com_twelvemonkeys_common_common-lang: common-lang 3.10.0
			com_twelvemonkeys_imageio_imageio-core: imageio-core 3.10.0
			com_twelvemonkeys_imageio_imageio-metadata: imageio-metadata 3.10.0
			com_twelvemonkeys_imageio_imageio-webp: imageio-webp 3.10.0
			org_quiltmc_parsers_gson: gson 0.2.1
			org_quiltmc_parsers_json: json 0.2.1
		yosbr: YOSBR 0.1.2
		yungsapi: YUNG's API 1.20-Fabric-4.0.5
		yungsbridges: YUNG's Bridges 1.20-Fabric-4.0.3
		yungsextras: YUNG's Extras 1.20-Fabric-4.0.3

**Logs:**
Description: Exception in server tick loop

java.lang.NoClassDefFoundError: net/minecraft/class_3999
	at com.ordana.spelunkery.events.ModEvents.polishingRecipe(ModEvents.java:213)
	at com.ordana.spelunkery.events.ModEvents.onBlockCLicked(ModEvents.java:79)
	at com.ordana.spelunkery.fabric.SpelunkeryFabric.onRightClickBlock(SpelunkeryFabric.java:46)
	at net.fabricmc.fabric.api.event.player.UseBlockCallback.lambda$static$0(UseBlockCallback.java:41)
	at net.minecraft.class_3225.handler$cci000$fabric-events-interaction-v0$interactBlock(class_3225.java:1587)
	at net.minecraft.class_3225.method_14262(class_3225.java)
	at net.minecraft.class_3244.method_12046(class_3244.java:1140)
	at net.minecraft.class_2885.method_12547(class_2885.java:34)
	at net.minecraft.class_2885.method_11054(class_2885.java:8)
	at net.minecraft.class_2600.method_11072(class_2600.java:22)
	at net.minecraft.class_3738.run(class_3738.java:18)
	at net.minecraft.class_1255.method_18859(class_1255.java:156)
	at net.minecraft.class_4093.method_18859(class_4093.java:23)
	at net.minecraft.server.MinecraftServer.method_24306(MinecraftServer.java:782)
	at net.minecraft.server.MinecraftServer.method_18859(MinecraftServer.java:164)
	at net.minecraft.class_1255.method_16075(class_1255.java:130)
	at net.minecraft.server.MinecraftServer.method_20415(MinecraftServer.java:17818)
	at net.minecraft.server.MinecraftServer.method_16075(MinecraftServer.java:758)
	at net.minecraft.class_1255.method_5383(class_1255.java:115)
	at net.minecraft.server.MinecraftServer.method_16208(MinecraftServer.java:742)
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:675)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:265)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.lang.ClassNotFoundException: net.minecraft.class_3999
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:226)
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	... 23 more

**To Reproduce:**

1. Take an unpolish diamond
2. polish it
3. crash!

**Expected behavior:**
not crashing

it only crashes on server, not on client

the only trace I have is this:
[22:11:37] [Server thread/ERROR]: Encountered an unexpected exception
java.lang.NoClassDefFoundError: net/minecraft/class_3999
	at com.ordana.spelunkery.events.ModEvents.polishingRecipe(ModEvents.java:213) ~[spelunkery-1.20.1-0.3.6-fabric.jar:?]
	at com.ordana.spelunkery.events.ModEvents.onBlockCLicked(ModEvents.java:79) ~[spelunkery-1.20.1-0.3.6-fabric.jar:?]
	at com.ordana.spelunkery.fabric.SpelunkeryFabric.onRightClickBlock(SpelunkeryFabric.java:46) ~[spelunkery-1.20.1-0.3.6-fabric.jar:?]
	at net.fabricmc.fabric.api.event.player.UseBlockCallback.lambda$static$0(UseBlockCallback.java:41) ~[fabric-events-interaction-v0-0.62+1802ada577-afa034351e41ee8.jar:?]
	at net.minecraft.class_3225.handler$cci000$fabric-events-interaction-v0$interactBlock(class_3225.java:1587) ~[server-intermediary.jar:?]
	at net.minecraft.class_3225.method_14262(class_3225.java) ~[server-intermediary.jar:?]
	at net.minecraft.class_3244.method_12046(class_3244.java:1140) ~[server-intermediary.jar:?]
	at net.minecraft.class_2885.method_12547(class_2885.java:34) ~[server-intermediary.jar:?]
	at net.minecraft.class_2885.method_11054(class_2885.java:8) ~[server-intermediary.jar:?]
	at net.minecraft.class_2600.method_11072(class_2600.java:22) ~[server-intermediary.jar:?]
	at net.minecraft.class_3738.run(class_3738.java:18) ~[server-intermediary.jar:?]
	at net.minecraft.class_1255.method_18859(class_1255.java:156) ~[server-intermediary.jar:?]
	at net.minecraft.class_4093.method_18859(class_4093.java:23) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_24306(MinecraftServer.java:782) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_18859(MinecraftServer.java:164) ~[server-intermediary.jar:?]
	at net.minecraft.class_1255.method_16075(class_1255.java:130) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_20415(MinecraftServer.java:17818) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_16075(MinecraftServer.java:758) ~[server-intermediary.jar:?]
	at net.minecraft.class_1255.method_18857(class_1255.java:139) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_16208(MinecraftServer.java:743) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:675) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:265) ~[server-intermediary.jar:?]
	at java.lang.Thread.run(Thread.java:840) ~[?:?]
Caused by: java.lang.ClassNotFoundException: net.minecraft.class_3999
	at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[?:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:226) ~[fabric-loader-0.16.0.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119) ~[fabric-loader-0.16.0.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[?:?]
	... 23 more
	
	
maybe its something only the client should execute and the server execute it too ?
It only happens when I use the wrong grindstone ex: normal grindstone with diamond.
I can try with less mods

I tested with less mods, the server still crashes when using the wrong grindstone for polishing

## Crashlogs
---- Minecraft Crash Report ----
// My bad.

Time: 2024-08-16 15:13:09
Description: Exception in server tick loop

java.lang.NoClassDefFoundError: net/minecraft/class_703
	at com.ordana.spelunkery.events.ModEvents.polishingRecipe(ModEvents.java:213)
	at com.ordana.spelunkery.events.ModEvents.onBlockCLicked(ModEvents.java:79)
	at com.ordana.spelunkery.fabric.SpelunkeryFabric.onRightClickBlock(SpelunkeryFabric.java:46)
	at net.fabricmc.fabric.api.event.player.UseBlockCallback.lambda$static$0(UseBlockCallback.java:41)
	at net.minecraft.class_3225.handler$zck000$fabric-events-interaction-v0$interactBlock(class_3225.java:587)
	at net.minecraft.class_3225.method_14262(class_3225.java)
	at net.minecraft.class_3244.method_12046(class_3244.java:1140)
	at net.minecraft.class_2885.method_12547(class_2885.java:34)
	at net.minecraft.class_2885.method_11054(class_2885.java:8)
	at net.minecraft.class_2600.method_11072(class_2600.java:22)
	at net.minecraft.class_3738.run(class_3738.java:18)
	at net.minecraft.class_1255.method_18859(class_1255.java:156)
	at net.minecraft.class_4093.method_18859(class_4093.java:23)
	at net.minecraft.server.MinecraftServer.method_24306(MinecraftServer.java:782)
	at net.minecraft.server.MinecraftServer.method_18859(MinecraftServer.java:164)
	at net.minecraft.class_1255.method_16075(class_1255.java:130)
	at net.minecraft.server.MinecraftServer.method_20415(MinecraftServer.java:764)
	at net.minecraft.server.MinecraftServer.method_16075(MinecraftServer.java:758)
	at net.minecraft.class_1255.method_18857(class_1255.java:139)
	at net.minecraft.server.MinecraftServer.method_16208(MinecraftServer.java:743)
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:675)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:265)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.lang.ClassNotFoundException: net.minecraft.class_703
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:226)
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	... 23 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- System Details --
Details:
	Minecraft Version: 1.20.1
	Minecraft Version ID: 1.20.1
	Operating System: Linux (amd64) version 5.10.0-21-amd64
	Java Version: 17.0.11, Eclipse Adoptium
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode, sharing), Eclipse Adoptium
	Memory: 402307072 bytes (383 MiB) / 1257242624 bytes (1199 MiB) up to 2097152000 bytes (2000 MiB)
	CPUs: 2
	Processor Vendor: AuthenticAMD
	Processor Name: AMD Ryzen 9 3950X 16-Core Processor
	Identifier: AuthenticAMD Family 23 Model 113 Stepping 0
	Microarchitecture: Zen 2
	Frequency (GHz): -0.00
	Number of physical packages: 1
	Number of physical CPUs: 16
	Number of logical CPUs: 32
	Graphics card #0 name: unknown
	Graphics card #0 vendor: unknown
	Graphics card #0 VRAM (MB): 0.00
	Graphics card #0 deviceId: unknown
	Graphics card #0 versionInfo: unknown
	Virtual memory max (MB): 165590.97
	Virtual memory used (MB): 138677.58
	Swap memory total (MB): 101183.99
	Swap memory used (MB): 43850.70
	JVM Flags: 3 total; -XX:+UseContainerSupport -Xms128M -Xmx2000M
	Fabric Mods: 
		fabric-api: Fabric API 0.92.2+1.20.1
			fabric-api-base: Fabric API Base 0.4.31+1802ada577
			fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.6.36+1802ada577
			fabric-biome-api-v1: Fabric Biome API (v1) 13.0.13+1802ada577
			fabric-block-api-v1: Fabric Block API (v1) 1.0.11+1802ada577
			fabric-block-view-api-v2: Fabric BlockView API (v2) 1.0.1+1802ada577
			fabric-command-api-v1: Fabric Command API (v1) 1.2.34+f71b366f77
			fabric-command-api-v2: Fabric Command API (v2) 2.2.13+1802ada577
			fabric-commands-v0: Fabric Commands (v0) 0.2.51+df3654b377
			fabric-containers-v0: Fabric Containers (v0) 0.1.64+df3654b377
			fabric-content-registries-v0: Fabric Content Registries (v0) 4.0.11+1802ada577
			fabric-convention-tags-v1: Fabric Convention Tags 1.5.5+1802ada577
			fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.2.19+1802ada577
			fabric-data-attachment-api-v1: Fabric Data Attachment API (v1) 1.0.0+de0fd6d177
			fabric-data-generation-api-v1: Fabric Data Generation API (v1) 12.3.4+1802ada577
			fabric-dimensions-v1: Fabric Dimensions API (v1) 2.1.54+1802ada577
			fabric-entity-events-v1: Fabric Entity Events (v1) 1.6.0+1c78457f77
			fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.6.2+1802ada577
			fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.63+df3654b377
			fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.40+1802ada577
			fabric-item-api-v1: Fabric Item API (v1) 2.1.28+1802ada577
			fabric-item-group-api-v1: Fabric Item Group API (v1) 4.0.12+1802ada577
			fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 2.2.22+1802ada577
			fabric-loot-api-v2: Fabric Loot API (v2) 1.2.1+1802ada577
			fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.1.45+9e7660c677
			fabric-message-api-v1: Fabric Message API (v1) 5.1.9+1802ada577
			fabric-mining-level-api-v1: Fabric Mining Level API (v1) 2.1.50+1802ada577
			fabric-networking-api-v1: Fabric Networking API (v1) 1.3.11+1802ada577
			fabric-networking-v0: Fabric Networking (v0) 0.3.51+df3654b377
			fabric-object-builder-api-v1: Fabric Object Builder API (v1) 11.1.3+1802ada577
			fabric-particles-v1: Fabric Particles (v1) 1.1.2+1802ada577
			fabric-recipe-api-v1: Fabric Recipe API (v1) 1.0.21+1802ada577
			fabric-registry-sync-v0: Fabric Registry Sync (v0) 2.3.3+1802ada577
			fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.3.37+92a0d36777
			fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 3.0.28+1802ada577
			fabric-resource-conditions-api-v1: Fabric Resource Conditions API (v1) 2.3.8+1802ada577
			fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.11.10+1802ada577
			fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.3.30+1802ada577
			fabric-transfer-api-v1: Fabric Transfer API (v1) 3.3.5+8dd72ea377
			fabric-transitive-access-wideners-v1: Fabric Transitive Access Wideners (v1) 4.3.1+1802ada577
		fabricloader: Fabric Loader 0.15.11
			mixinextras: MixinExtras 0.3.5
		java: OpenJDK 64-Bit Server VM 17
		memoryleakfix: Memory Leak Fix 1.1.5
		minecraft: Minecraft 1.20.1
		moonlight: Moonlight 1.20-2.12.11
		spelunkery: Spelunkery 1.20.1-0.3.8
	Server Running: true
	Player Count: 1 / 20; [class_3222['ErasedTruth'/387, l='ServerLevel[world]', x=-88.09, y=64.00, z=-154.54]]
	Data Packs: vanilla, Spelunkery Generated Pack, fabric
	Enabled Feature Flags: minecraft:vanilla
	World Generation: Experimental
	Is Modded: Definitely; Server brand changed to 'fabric'
	Type: Dedicated Server (map_server.txt)

--- a/common/src/main/java/com/ordana/spelunkery/events/ModEvents.java
+++ b/common/src/main/java/com/ordana/spelunkery/events/ModEvents.java
@@ -210,8 +210,11 @@ public class ModEvents {
                         if (!player.getAbilities().instabuild) stack.shrink(1);
                     }
                     else if (state.is(ModBlocks.DIAMOND_GRINDSTONE.get()) && state.getValue(ModBlockProperties.DEPLETION) == 3 && polishingRecipe.isRequiresDiamondGrindstone() || (polishingRecipe.isRequiresDiamondGrindstone() && !state.is(ModBlocks.DIAMOND_GRINDSTONE.get()))) {
-                        ParticleUtil.spawnParticlesOnBlockFaces(level, pos, ParticleTypes.SMOKE, UniformInt.of(3, 5), -0.05f, 0.05f, false);
-                        player.swing(hand);
+                        if (level.isClientSide()) {
+                            ParticleUtil.spawnParticlesOnBlockFaces(level, pos, ParticleTypes.SMOKE, UniformInt.of(3, 5), -0.05f, 0.05f, false);
+                            player.swing(hand);
+                        }
+
                         level.playSound(player, pos, SoundEvents.SHIELD_BREAK, SoundSource.BLOCKS, 0.5F, 0.0F);
                         return InteractionResult.sidedSuccess(level.isClientSide);
                     }

